http4k-testing/approval/src/main/kotlin/org/http4k/testing/Approver.kt

```diff
@@ -23,17 +23,24 @@ interface Approver {
      * Check the content of the passed message against the previously approved content.
      */
     fun <T : HttpMessage> assertApproved(httpMessage: T)
+
+    fun withNameSuffix(suffix: String): Approver
 }
 
 class NamedResourceApprover(
-    private val name: String,
+    private var name: String,
     private val approvalContent: ApprovalContent,
     private val approvalSource: ApprovalSource
 ) : Approver {
 
     private val matchAllLineEndings = "\\r\\n?".toRegex()
     private fun String.normalizeLineEndings() = replace(matchAllLineEndings, "\n")
 
+    override fun withNameSuffix(suffix: String): Approver {
+        name = name + suffix
+        return this
+    }
```

Comment:
This will infinitely extend the filename name

```kotlin
approver.withNameSuffix("test1").hasApprovedContent(...) // result = "TestClass.testNametest1"
approver.withNameSuffix("test2").hasApprovedContent(...) // result = "TestClass.testNametest1test2"
```

Could we instead of returning `this`, return a a new copy of `NamedResourceApprover`? 

```kotlin
    override fun withNameSuffix(suffix: String): Approver {
        return NamedResourceApprover( name = name + suffix, ... )
    }
```

Returning a new copy means that the base name stays the same, while the suffixes change. 

The result would be the following files in your test. 

Current: 

```
ExampleApprovalTest.dynamic testA.approved
ExampleApprovalTest.dynamic testAB.approved
ExampleApprovalTest.dynamic testABC.approved
```

Fixed: 

```
ExampleApprovalTest.dynamic testA.approved
ExampleApprovalTest.dynamic testB.approved
ExampleApprovalTest.dynamic testC.approved
```

Perhaps we can also do `name = "$name.$suffix"` to separate the basename from the suffix?